### PR TITLE
[using-page-transitions] fixes #2669

### LINKED ISSUE #2669: [using-page-transitions] build fails
```console
error Building static HTML for pages failed

See our docs page on debugging HTML builds for help https://goo.gl/yL9lND

  4 | 
  5 | const Cat = ({ transition }) => (
> 6 |   <div style={transition.style}>
    |                          ^
  7 |     <h1>meow</h1>
  8 |     <div>
  9 |       <Link to="/">Go to home</Link>


  WebpackError: Cannot read property 'style' of undefined
  
  - cat.js:6 Cat
    src/pages/cat.js:6:26
  
  - ReactCompositeComponent.js:306 ReactCompositeComponentWrapper._constructComponentWithoutOwner
    ~/react-dom/lib/ReactCompositeComponent.js:306:1
  
  - ReactCompositeComponent.js:282 ReactCompositeComponentWrapper._constructComponent
    ~/react-dom/lib/ReactCompositeComponent.js:282:1
  
  - ReactCompositeComponent.js:185 ReactCompositeComponentWrapper.mountComponent
    ~/react-dom/lib/ReactCompositeComponent.js:185:1
  
  - ReactReconciler.js:43 Object.mountComponent
    ~/react-dom/lib/ReactReconciler.js:43:1
  
  - ReactMultiChild.js:234 ReactDOMComponent.mountChildren
    ~/react-dom/lib/ReactMultiChild.js:234:1
  
  - ReactDOMComponent.js:657 ReactDOMComponent._createContentMarkup
    ~/react-dom/lib/ReactDOMComponent.js:657:1
  
  - ReactDOMComponent.js:524 ReactDOMComponent.mountComponent
    ~/react-dom/lib/ReactDOMComponent.js:524:1
  
  - ReactReconciler.js:43 Object.mountComponent
    ~/react-dom/lib/ReactReconciler.js:43:1
  
  - ReactMultiChild.js:234 ReactDOMComponent.mountChildren
    ~/react-dom/lib/ReactMultiChild.js:234:1
```

[Looks like the preview may have failed in the PR too](https://github.com/gatsbyjs/gatsby/pull/2557#issuecomment-339480744)? Not exactly sure that's what's happening there.

A quick workaround was to swap:

```jsx
style={transition.style}
```

for:

```jsx
style={(typeof transition != 'undefined') ? transition.style : null}
```

but this check should happen just once elsewhere…haven't quite figured out where.

### Edit

That 'workaround' makes the build pass, but the resulting site isn't functional. When clicking 'local' links:

```console
gatsby-browser.js:21 Uncaught TypeError: Cannot read property 'component' of undefined
    at gatsby-browser.js:21
    at t.render (Transition.js:372)
    at p._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:796)
    at p._renderValidatedComponent (ReactCompositeComponent.js:819)
    at performInitialMount (ReactCompositeComponent.js:359)
    at p.mountComponent (ReactCompositeComponent.js:255)
    at Object.mountComponent (ReactReconciler.js:43)
    at Object.updateChildren (ReactChildReconciler.js:119)
    at v._reconcilerUpdateChildren (ReactMultiChild.js:209)
    at v._updateChildren (ReactMultiChild.js:308)
```

### PR DESCRIPTION
**Pull request by andykenward**
On a production build `pageResources` werent being passed through as a prop if you `replaceComponentRenderer` in`gatsby-browser.js`

### CODE REVIEW

File: packages/gatsby/cache-dir/component-renderer.js [hunk 1]
```diff
@@ -104,7 +104,7 @@ class ComponentRenderer extends React.Component {
 
   render() {
     const pluginResponses = apiRunner(`replaceComponentRenderer`, {
-      props: this.props,
+      props: { ...this.props, pageResources: { ...this.state.pageResources } },
```
Comment (KyleAMathews) [inline packages/gatsby/cache-dir/component-renderer.js]: this.state.pageResources doesn't need spread

Comment (andykenward) [reply packages/gatsby/cache-dir/component-renderer.js]: @KyleAMathews done and pushed

**Comment by KyleAMathews** [comment]
Thanks!

**Comment by jlengstorf** [comment]
Hiya @andykenward! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**